src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -17,7 +19,7 @@ internal abstract class SynthesizedEntryPointSymbol : MethodSymbol
         internal const string FactoryName = "<Factory>";
 
         private readonly NamedTypeSymbol _containingType;
-        private readonly TypeSymbol _returnType;
+        private TypeSymbol _returnType;
```

Comment:
>private TypeSymbol _returnType; [](start = 8, length = 31)

It feels like we could still have it readonly, it looks like we are calculating all the information in the constructor anyway. It is fine to follow up on thin in the next PR, I think. Please add a prototype comment.